include/walker-dropdown.php

```diff
@@ -31,12 +31,12 @@ class PLL_Walker_Dropdown extends Walker {
 	 * @return void
 	 */
 	public function start_el( &$output, $element, $depth = 0, $args = array(), $current_object_id = 0 ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		$value = $args['value'];
+		$value_type = $args['value'];
 		$output .= sprintf(
 			"\t" . '<option value="%1$s"%2$s%3$s>%4$s</option>' . "\n",
-			esc_attr( $element->$value ),
+			'slug' === $value_type ? esc_attr( $element->$value_type ) : esc_url( $element->$value_type ),
```

Comment:
You should check for `url` instead, as this is the new type of values that you are introducing which requires a specific escaping function `esc_url()`. As is, all types of values would be escaped by `esc_url()` except slugs. Here is an example of usage non-slug non-url value: https://github.com/polylang/polylang/blob/3.1/admin/admin-filters-term.php#L137

Reply:
Thanks, I didn't thought about other value than slug or url...